Parse DSDT Table with Signature Sharing Fixed

raisfathin opened the pull request:
Basically applying commits in [Add DSDT parsing support](https://github.com/raisfathin/chipsec/pull/1) to the latest upstream master.

Needed to resolve some merge conflicts, though.

raisfathin committed:
Add FADT table parsing support

raisfathin committed:
Parse DSDT table address in FADT

raisfathin committed:
Add tests for DSDT table address parsing

tweksteen commented on source/tool/chipsec/hal/acpi_tables.py:
```diff
@@ -584,3 +586,30 @@ def __str__( self ):
+
+    def parse( self, table_content ):
+        self.dsdt = struct.unpack('<I', table_content[4:8])[0]
+        try:
```
Any reason why we are not using a condition on the length of table_content to try to fill XSDT address?

raisfathin replied:
I thought it is more common in python to ask forgiveness instead of asking for permission? I don't have much experience in python, though :)

But sure, the existence of X_DSDT can be inferred from the length of the table (or the revision?). What do you think?

tweksteen replied:
Right now, it is not obvious why you are intercepting this exception type. I think the code would be clearer if you'd use a simple condition.

raisfathin replied:
Done.

tweksteen commented on source/tool/chipsec/hal/acpi.py:
```diff
@@ -371,14 +371,55 @@ def get_ACPI_table_list( self ):
+        x_dsdt_address = parsed_fadt_content.x_dsdt
+        dsdt_address_to_use = None
+
+        if x_dsdt_address is None:
```
I am wondering if this logic should be part of the FADT class. Maybe add a method to decide which SDT to use?

raisfathin replied:
Hmm, sure. It should make the logic easier to follow, too, I think.

raisfathin replied:
Done.

tweksteen commented on source/tool/tests/software/mock_helper.py:
```diff
@@ -130,6 +130,55 @@ def read_phys_mem(self, pa_hi, pa_lo, length):
+
+    def _ensure_fadt_in_sdt(self, SDT_ENTRIES):
+        if self.FADT_ADDRESS not in SDT_ENTRIES:
+            SDT_ENTRIES.append(self.FADT_ADDRESS)
```
SDT_ENTRIES is not a constant anymore at this stage. Could it be moved to an instance attribute?

raisfathin replied:
...Not sure. `_ensure_fadt_in_sadt` was written so that classes that derive from `DSDTParsingHelper` need not to worry about adding FADT entry to R/XSDT.

On a second thought, expecting the user to add an FADT entry to R/XSDT sounds better, because the user knows that they are changing a constant and that the constant is not going to be changed by anything else (therefore it's their responsibility to fill in the correct entries to R/XSDT).

By moving it to instance attribute, do you mean the derived classes need to add their needed entries to R/XSDT in their own `__init__` method?

tweksteen replied:
It is good to have a parent class to handle the FADT entry. The issue here is that SDT_ENTRIES is not a constant and it makes more sense to have it as an attribute. The same should have been done for RSDT_ENTRIES, XSDT_ENTRIES, RSDP_DESCRIPTOR, RSDT_DESCRIPTOR and XSDT_DESCRIPTOR in ACPIHelper. Since you are generating their content at execution, depending on the constant from the child classes, they should be attributes.

raisfathin replied:
Done, but not sure whether it's good enough. I avoided using `hasattr` because it doesn't look good :s

c7zero committed:
PCI option ROM enumeration and extraction

1. Added PCI expansion (option) ROM enumeration and extraction
capability
2. Added capability to dump entire legacy PCI config space for a given
or all available devices
3. Cosmetic changes in spd and cmos
4. Fixed register types in hsx.xml

c7zero committed:
UEFI parsing changes

1. Added ability to calculate hashes of EFI binaries
2. Added ability to search EFI binaries by UI name, GUID, hashes or
contents matching regular expressions
3. Refactored UEFI firmware image parsing
4. Fixed get_tools_path in helpers

c7zero committed:
UEFI image parsing enhancements

1. Automated identification of EFI NVRAM type (still cannot
auto-distinguish between various VSS types). NVRAM type is now optional
in all uefi util commands
2. Optimized performance of firmware volume parsing (it was taking
forever to parse FV with large chunk of F's)
3. Added auto-parsing of NVRAM when parsing firmware volumes with
variable store GUIDs

raisfathin commented:
Ready for another review!

c7zero committed:
Updated EFI module search

Added a mode to find all occurrences of EFI binaries matching search
criteria while searching within UEFI firmware image vs retuning upon the
first match (findall argument in traverse_uefi_region)

raisfathin commented:
Ready for another review!

tweksteen commented:
LGTM!
